### PR TITLE
Remove unneeded deps

### PR DESCRIPTION
**Pull request by margueritepd**
These were added to fix some compile issues which I cannot reproduce.
Additionally, having these lines causes the following errors when GoCD
tries to load the plugin:

```
2015-05-13 08:45:12,320 ERROR [Thread-65] FelixGoPluginOSGiFramework:124
- Failed to load plugin: plugins_work\gocd-hipchat-plugin.jar. Plugin is
  invalid. Reasons [
Class [scala.tools.ant.ClassloadVerify] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.ant.FastScalac] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.ant.Pack200Task] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.ant.Same] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.ant.Scalac] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.ant.Scaladoc] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.ant.ScalaMatchingTask] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.ant.ScalaTool] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.ant.sabbus.Break] could not be loaded. Message: [org/apache/tools/ant/Task].,
Class [scala.tools.ant.sabbus.Make] could not be loaded. Message: [org/apache/tools/ant/Task].,
Class [scala.tools.ant.sabbus.ScalacFork] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.ant.sabbus.Use] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
Class [scala.tools.nsc.interpreter.JLineDelimiter] could not be loaded.  Message: [scala/tools/jline/console/completer/ArgumentCompleter$ArgumentDelimiter].,
Class [scala.tools.nsc.interpreter.JLineReader$$anon$1] could not be loaded. Message: [scala/tools/jline/console/completer/Completer].,
Class [scala.tools.nsc.interpreter.JLineReader$JLineConsoleReader] could not be loaded. Message: [scala/tools/jline/console/ConsoleReader].,
Class [scala.tools.nsc.interpreter.session.FileBackedHistory] could not be loaded. Message: [scala/tools/jline/console/history/History].,
Class [scala.tools.nsc.interpreter.session.JLineHistory$JLineFileHistory] could not be loaded. Message: [scala/tools/jline/console/history/History].,
Class [scala.tools.nsc.interpreter.session.JLineHistory] could not be loaded.  Message: [scala/tools/jline/console/history/History].,
Class [scala.tools.nsc.interpreter.session.SimpleHistory$Entry] could not be loaded. Message: [scala/tools/jline/console/history/History$Entry].,
Class [scala.tools.nsc.interpreter.session.SimpleHistory] could not be loaded. Message: [scala/tools/jline/console/history/History].
]
```

(See https://github.com/PagerDuty/gocd-hipchat-plugin/issues/7 )

So let's remove these lines to see if that fixes the issue or causes any
other issues!

### CODE REVIEW

**Comment by ranjib** [comment]
:+1: 
may be @danmbyrd @dmyang can shed some light ..

**Comment by danmbyrd** [comment]
Here is my theory for what is going on here. Lets start with the error messages:

```
Class [scala.tools.ant.ClassloadVerify] could not be loaded. Message: [org/apache/tools/ant/taskdefs/MatchingTask].,
```

I'm interpreting this to mean that the JVM attempted to load the former class (```scala.tools.ant.ClassloadVerify```), but failed, as it was unable to load the later class (```org/apache/tools/ant/taskdefs/MatchingTask```).

Looking at the assembled jar, I can see that the classes on the right are all absent, and are contained in two jars that are not on the compile or runtime time classpath (org.scala-lang:jline:2.10.4 and the [ant jar](https://github.com/apache/ant/blob/master/src/main/org/apache/tools/ant/taskdefs/MatchingTask.java)). Whats strange is that I wouldn't expect these classes to be loaded by the jvm, as they all seem unrelated. One solution might be to add them as dependencies (the jline jar is an [optional dependency](http://central.maven.org/maven2/org/scala-lang/scala-compiler/2.10.0/scala-compiler-2.10.0.pom) of scala-compiler, so ```compile->*``` is telling sbt to fetch it even though its optional):

```scala
libraryDependencies ++= Seq(
  "org.scala-lang" % "scala-compiler" % scalaVersion.value % "compile->*",
  "org.apache.ant" % "ant" % "1.9.6",
  "org.scalaj" %% "scalaj-http" % "1.0.1",
  "org.json4s" %% "json4s-native" % "3.2.11",
  "cd.go.plugin" %  "go-plugin-api" % goVersion
)
```
Alternatively change how this jar is being loaded at the call site, which seems to be causing all these classes to be loaded, might help.

File: build.sbt [hunk 1]
```diff
@@ -19,9 +19,6 @@ assemblyJarName in assembly := name.value + ".jar"
 scalariformSettings
 
 libraryDependencies ++= Seq(
-  "org.scala-lang" % "scala-compiler" % scalaVersion.value,
-  "org.scala-lang" % "scala-library" % scalaVersion.value,
-  "org.scala-lang" % "scala-reflect" % scalaVersion.value,
   "org.scalaj" %% "scalaj-http" % "1.0.1",
   "org.json4s" %% "json4s-native" % "3.2.11" exclude("org.scala-lang", "scalap"),
   "com.thoughtworks.go" %%  "go-plugin-api" % "current"  % "provided" from "https://bintray.com/artifact/download/gocd/gocd/go-plugin-api-14.4.0.jar"
```
Comment (danmbyrd) [inline build.sbt]: This is being properly published now, so you can get it more canonically with:

```scala
"cd.go.plugin" %  "go-plugin-api" % goVersion,
```

File: build.sbt [hunk 2]
```diff
@@ -19,9 +19,6 @@ assemblyJarName in assembly := name.value + ".jar"
 scalariformSettings
 
 libraryDependencies ++= Seq(
-  "org.scala-lang" % "scala-compiler" % scalaVersion.value,
-  "org.scala-lang" % "scala-library" % scalaVersion.value,
-  "org.scala-lang" % "scala-reflect" % scalaVersion.value,
   "org.scalaj" %% "scalaj-http" % "1.0.1",
   "org.json4s" %% "json4s-native" % "3.2.11" exclude("org.scala-lang", "scalap"),
```
Comment (danmbyrd) [inline build.sbt]: Not sure why this ```exclude("org.scala-lang", "scalap")``` was originally added, don't see why it is needed.

Comment (margueritepd) [inline build.sbt]: Not sure why excluded. The `json4s` README has the following under a list of jars to download:
> scalap (Only for Scala 2.10, 2.11 compatible versions)

May be related?

**Comment by margueritepd** [comment]
@danmbyrd 
From an earlier comment

> I'm interpreting this to mean that the JVM attempted to load the former class (scala.tools.ant.ClassloadVerify), but failed, as it was unable to load the later class (org/apache/tools/ant/taskdefs/MatchingTask). For this reason, I'm not confident that removing those extra dependency is going to help the situation.

Now that we know that gocd is explicitly loading all classes in the class path, does this reason still hold?

**Comment by Ehekatl** [comment]
I'm using gocd v15.2, and delete these three lines let this plugin works
Thanks a lot

**Comment by margueritepd** [comment]
This should fix the load errors.

However it turns out we were also building the jar incorrectly. We should be putting dependencies in the `lib/` directory, according to the instructions on how to build a gocd plugin. So we should fix that too

**Pull request merged**